go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -851,7 +915,19 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 	case factory.EgressNodeType:
 		oldNode := oldObj.(*kapi.Node)
 		newNode := newObj.(*kapi.Node)
-
+		// Add the node to the tracker only if it belongs to the local zone.
+		// TODO: Find a better place to put this. See if LoadOrStore has any perf impact if its called too frequently
+		klog.Infof("SURYA %v", config.OVNKubernetesFeature.EnableInterconnect)
+		if config.OVNKubernetesFeature.EnableInterconnect {
+			// handle zone annotation update on the node
+			if h.oc.isLocalZoneNode(newNode) {
```

Comment:
when a node moves zone, should we tear setup here and let the new master redo the setup? wondering how frequently a zone move can happen.